### PR TITLE
[Fix](Json type) correct cast result for json type

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes

1. datatype->to_string result in incorrect string value, use serdes instead
2. cast json int to json tinyint should not result null

Will result behavior change

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/functions/function_cast.h [hunk 1]
```diff
@@ -845,48 +860,21 @@ struct ConvertImplFromJsonb {
...
-                    if (value->isInt8() || value->isInt16() || value->isInt32() ||
-                        value->isInt64() || value->isInt128()) {
-                        res[i] = (int128_t)((const JsonbIntVal*)value)->val();
+                } else if constexpr (type_index == TypeIndex::Int8 ||
+                                     type_index == TypeIndex::Int16 ||
+                                     type_index == TypeIndex::Int32 ||
+                                     type_index == TypeIndex::Int64 ||
+                                     type_index == TypeIndex::Int128) {
+                    if (value->isInt()) {
+                        res[i] = ((const JsonbIntVal*)value)->val();
```
Comment (xiaokang) [inline be/src/vec/functions/function_cast.h:869]: Will it cause wrong result, eg int64 -> int8 ?

**Review by xiaokang — APPROVED** [review]
LGTM

File: be/src/vec/functions/function_cast.h (hunk 1 above)
Comment (eldenmoon) [reply be/src/vec/functions/function_cast.h:869]: the same as cast bigint to tinyint

**Review by qidaye — APPROVED** [review]

**Comment by eldenmoon** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**